### PR TITLE
Extend support for clusters (Python only)

### PR DESCRIPTION
**Pull request by RHamalainen**
Hello,

This is essentially the same content as in pull request #132 but without formatting changes.

No changes to required Python version compared to original code base.

#### Example 1: modify

```
"PERIPHERAL_*":
  _modify:
    "REG_*1":
      addressOffset: 8
  _clusters:
    CLU_*:
      _modify:
        REG_2:
          addressOffset: 0
    CLU_2:
      _modify:
        REG_1:
          name: my_REG_1
```

#### Example 2: add

```
"PERIPHERAL_*":
  _add:
    "REG_COMMON_NEW":
      addressOffset: 8
  _clusters:
    CLU_*:
      _add:
        REG_NEW:
          addressOffset: 8
    _add:
      "CLU_3":
        addressOffset: 300
        "REG_1":
          addressOffset: 0
        "REG_2":
          addressOffset: 4
```

#### Example 3: delete

```
"PERIPHERAL_*":
  _delete:
    - "REG_*1"
  _clusters:
    "CLU_*":
      _delete:
        - "REG_*1"
```

#### Example 4: strip

```
"PERIPHERAL_*":
  _strip:
    - "REG_*"
  _clusters:
    "CLU_*":
      _strip:
        - "R*"
    "CLU_1":
      _strip_end:
        - "1"
```

Best regards,

RHamalainen

### CODE REVIEW

**Comment by burrbull** [comment]
Updated python to 3.9. Rebase, please

**Comment by RHamalainen** [comment]
Hello,

New changes merged.

Best regards,

RHamalainen

**Comment by burrbull** [comment]
Nooooo.
No merge. Rebase only.

**Comment by RHamalainen** [comment]
Oops. Mistake fixed.

**Comment by burrbull** [comment]
format with `black`, please

**Comment by burrbull** [comment]
Could you also test #138 ?

**Comment by RHamalainen** [comment]
Formatted with black.

**Comment by burrbull** [comment]
Possibly we need to add diff there
https://github.com/stm32-rs/svdtools/blob/ce6ff6663b752d6ed8732432c36008b4386eb7d0/.github/workflows/ci.yaml#L29
```
black --check --diff svdtools
```

**Comment by burrbull** [comment]
Now when I look at this again: `Example 2: add` proposal is not consistent with current format.
It should be something like:
```
"PERIPHERAL_*":
  _add:
    "REG_COMMON_NEW":
      addressOffset: 8
    _clusters:
      "CLU_3":
        addressOffset: 300
        "REG_1":
          addressOffset: 0
        "REG_2":
          addressOffset: 4
  _clusters:
    CLU_*:
      _add:
        REG_NEW:
          addressOffset: 8
```
I mean "_add:_clusters" instead of "_clusters:_add"  to add something inside peripheral.
As it is already done for "_add:_registers" (default for "_add") and "_add:_interrupts".

**Comment by burrbull** [comment]
```yaml
"PERIPHERAL_*":
  _add:
    "REG_COMMON_NEW":
      addressOffset: 8
    _clusters:
      "CLU_3":
        addressOffset: 300
        registers:
            "REG_1":
              addressOffset: 0
            "REG_2":
              addressOffset: 4
  _clusters:
    CLU_*:
      _add:
        REG_NEW:
          addressOffset: 8
```

File: svdtools/patch.py [hunk 1]
```diff
@@ -737,12 +769,19 @@ class Peripheral:
...
+
+        Ignore `register`s owned by `cluster`s.
         """
         for rtag in self.ptag.iter("register"):
+            parent = rtag.getparent()
+            assert isinstance(parent, Element), "Register must have parent."
+            if parent.tag == "cluster":
+                continue
             name = rtag.find("name").text
             if matchname(name, rspec):
```
Comment (burrbull) [inline svdtools/patch.py:786]: cc @adamgreig What do you think about this breaking change?

Previously it was an equivalent of `all_registers` of peripheral.

Comment (burrbull) [reply svdtools/patch.py:786]: Should we move this in separate PR synchronized with same change in rust version?

**Pull request merged**